src/librustdoc/html/highlight.rs

```diff
@@ -144,14 +189,19 @@ impl Iterator for TokenIter<'a> {
     }
 }
 
-fn get_real_ident_class(text: &str, edition: Edition) -> Class {
-    match text {
+/// Classifies into identifier class; returns `None` if this is a non-keyword identifier.
+fn get_real_ident_class(text: &str, edition: Edition, allow_path_keywords: bool) -> Option<Class> {
+    let ignore: &[&str] =
+        if allow_path_keywords { &["self", "Self", "super", "crate"] } else { &["self", "Self"] };
+    if ignore.iter().any(|k| *k == text) {
+        return None;
```

Comment:
Isn't this backwards? Shouldn't it return Some(_) for `super` when paths are allowed?

Reply:
I think it'd be worth improving the documentation on this function or maybe split it into two functions. Explanations: it is used in two contexts: one where we want to get the whole path (where `super` and `self` can be part of the path) and the other is in case the path is the keyword itself (so only "super" of "crate", etc).

So with this information, what do you prefer? Two functions or simply extending the documentation?